leveldown.js

```diff
@@ -82,25 +105,55 @@ SubDown.prototype.type = 'subleveldown'
 SubDown.prototype._open = function (opts, cb) {
   var self = this
 
-  this.db.open(function (err) {
-    if (err) return cb(err)
-
-    var subdb = reachdown(self.db, 'subleveldown')
+  if (this.leveldown.status === 'open') {
+    process.nextTick(finish)
+  } else if (this.leveldown.status === 'opening') {
+    if (!this._externalOpener) {
+      return process.nextTick(cb, new Error('Database was opened by third party'))
```

Comment:
Needs a better error message. The expectation is that the user either passes in:

- a fresh `levelup` db, in which case `levelup` has at this point already started to open, and we need to hook into that because `abstract-leveldown`  doesn't have open/close events yet
- an `abstract-leveldown` db, without deferred open (meaning the db is at this point either closed or open, not opening or closing)